Manually adds VISION_OS to Platform to work around OpenAPI spec issue

mathiasemil opened the pull request:
The OpenAPI spec 2.4 and 3.0 fails include VISION_OS in Platform. This breaks several endpoints since the API is returning VISION_OS for all apps that support VisionOS.

Example from https://api.appstoreconnect.apple.com/v1/appInfos/xxx:
```json
"platforms" : [ "IOS", "MAC_OS", "TV_OS", "VISION_OS" ]
```

Feedback filed as FB13208526.

I'm unsure how to handle this without manually modifying Platform.swift and I realize that this "fix" will be undone next time a new version of the OpenAPI spec is generated (unless Apple fixes it in that version).

Any thoughts for a better way to handle this?

mathiasemil commented on Sources/OpenAPI/Generated/Entities/Platform.swift:
```diff
@@ -9,4 +9,5 @@ public enum Platform: String, Codable, CaseIterable {
 	case ios = "IOS"
 	case macOs = "MAC_OS"
 	case tvOs = "TV_OS"
+    case visionOs = "VISION_OS" // VisionOS manually added since version 3.0 of Apple's OpenAPI spec fails to include it. Reported in FB13208526 27/9/2023.
```
I'm unsure why GitHub shows the added line with a different indentation, but in Xcode it looks correct 🤷🏼

MortenGregersen commented:
I saw the same issue when I updated https://github.com/MortenGregersen/Bagbutik to version 3.0 of the spec. I have reported this as FB13071298. Apple has replied on my issue (September 19th):
> *We agree the API should say nothing about unreleased feature.*
And they are looking into publishing a new spec (sometime).

Maybe you should make yours as duplicate of mine? I will update mine to reference yours.

mathiasemil commented:
> I saw the same issue when I updated https://Bagbutik.dev to version 3.0 of the spec ([MortenGregersen/Bagbutik#131](https://github.com/MortenGregersen/Bagbutik/pull/131)).
> 
> I have reported this as FB13071298. Apple has replied on my issue (September 19th):
> 
> > _We agree the API should say nothing about unreleased feature._
> > And they are looking into publishing a new spec (sometime).
> 
> Maybe you should make yours as duplicate of mine? I will update mine to reference yours.

I've updated my feedback now to reference yours 🙏🏼

AvdLee reviewed (approved):
> Any thoughts for a better way to handle this?

Ideally, we would apply a custom parsing rule to CreateAPI, but this will do for now. Hopefully, it gets fixed by Apple soon!
